libraries/Microsoft.Bot.Builder/CloudAdapterBase.cs

```diff
@@ -242,18 +204,23 @@ protected async Task ProcessProactiveAsync(ClaimsIdentity claimsIdentity, Conver
             // Use the cloud environment to create the credentials for proactive requests.
             var proactiveCredentialsResult = await _botFrameworkAuthentication.GetProactiveCredentialsAsync(claimsIdentity, audience, cancellationToken).ConfigureAwait(false);
 
+            // Create a ConnectorFactory instance for the application to create new ClientConnector instances that can call alternative endpoints using credentials for the current appId. 
+            var connectorFactory = new CloudAdapterConnectorFactory(this, claimsIdentity);
+
             // Create the connector client to use for outbound requests.
             using (var connectorClient = new ConnectorClient(new Uri(reference.ServiceUrl), proactiveCredentialsResult.Credentials, _httpClient, disposeHttpClient: _httpClient == null))
```

Comment:
Why doesn't this use the method in the factory created a couple lines prior?

Reply:
It could - I thought this was clearer as it was more explicit - but I guess that would funnel all the clear calls through the same code (i.e. through the factory)

So you think I should change it?

Reply:
@EricDahlvang it does now - a slightly bigger change than you might have thought - I make the authentication class the source for the client factory and used the factory. it seems to simplify things ultimately.